devices/debian_wifi.py

```diff
@@ -1,88 +1,126 @@
-import re
+import re, time
 import debian
 import pexpect
+from countrycode import countrycode
+from lib.wifi import wifi_client_stub
+import pdb
 
-class DebianWifi(debian.DebianBox):
+class DebianWifi(debian.DebianBox, wifi_client_stub):
     '''Extension of Debian class with wifi functions'''
 
     model = ('debianwifi')
+    wifi_interface = "wlan0"
+    wifi_interface1 = "wlan1"
 
-    iface_wlan = "wlan1"
+    wifi_client_control = None
 
-    def scan(self):
+    def disable_and_enable_wifi(self, iface):
+       self.set_link_state(iface,"down")
+       self.set_link_state(iface,"up")
+
+    def wifi_scan(self, iface, ssid_name=None):
         from tests.lib.installers import install_iw
         install_iw(self)
 
-        self.sendline('iw %s scan | grep SSID:' % self.iface_wlan)
-        self.expect(self.prompt)
-
+        self.sudo_sendline('iw %s scan | grep SSID:' % iface)
+        idx = self.expect(["password for .*:"] + self.prompt,timeout=20)
```

Comment:
I think you missed the entire point of sudo_sendline(). It's supposed to handle all aspects of the password prompt.

Reply:
Will change this